examples/pcr/extend.c

```diff
@@ -0,0 +1,218 @@
+/* extend.c
+ *
+ * Copyright (C) 2006-2020 wolfSSL Inc.
+ *
+ * This file is part of wolfTPM.
+ *
+ * wolfTPM is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfTPM is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+/* This is a helper tool for extending hash into a TPM2.0 PCR */
+
+#include <wolftpm/tpm2_wrap.h>
+
+#ifndef WOLFTPM2_NO_WOLFCRYPT
+#include <wolfssl/wolfcrypt/hash.h>
+#endif
+
+#include <examples/pcr/extend.h>
+#include <examples/tpm_io.h>
+#include <examples/tpm_test.h>
+
+#include <stdio.h>
+#include <stdlib.h> /* strtol */
+
+
+/******************************************************************************/
+/* --- BEGIN TPM2.0 PCR Extend example tool  -- */
+/******************************************************************************/
+
+static void usage(void)
+{
+    printf("Expected usage:\n");
+    printf("./examples/pcr/extend [pcr] [filename]\n");
+    printf("* pcr is a PCR index between 0-23 (default 16)\n");
+    printf("* filename points to file(data) to measure\n");
+    printf("\tIf wolfTPM is built with --disable-wolfcrypt the file\n"
+           "\tmust contain SHA256 digest ready for extend operation.\n"
+           "\tOtherwise, the extend tool computes the hash using wolfcrypt.\n");
+    printf("Demo usage without parameters, extends PCR16 with known hash.\n");
+}
+
+int TPM2_Extend_Test(void* userCtx, int argc, char *argv[])
+{
+    int i, pcrIndex, len, rc = -1;
+    WOLFTPM2_DEV dev;
+    /* Arbitrary user data provided through a file */
+    const char *filename = NULL;
+    FILE *dataFile = NULL;
+    unsigned char hash[TPM_SHA256_DIGEST_SIZE];
```

Comment:
For consistency you could use `byte` here. It's always available and used for all wolf libs.